ts/components/CodeInput.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  CodeInput as IOCodeInput,
+  WithTestID
+} from "@pagopa/io-app-design-system";
+import { useFocusEffect } from "@react-navigation/native";
+import React from "react";
+import { StyleSheet, View, TextInput } from "react-native";
+import { Millisecond } from "@pagopa/ts-commons/lib/units";
+import { setAccessibilityFocus } from "../utils/accessibility";
+
+type CodeInputProps = WithTestID<Parameters<typeof IOCodeInput>[0]> & {
+  onChange: (text: string) => void;
+  accessibilityLabel: string;
+  onFocusedA11yView?: React.RefObject<View>;
+};
+
+const FOCUS_TIMEOUT = 1000;
+const A11Y_FOCUS_TIMEOUT = FOCUS_TIMEOUT + 200;
+const FOCUS_TIMEOUT_MS = FOCUS_TIMEOUT as Millisecond;
+const A11Y_FOCUS_TIMEOUT_MS = A11Y_FOCUS_TIMEOUT as Millisecond;
+
+/**
+ * Temporary wrapper to DS CodeInput to allow keyboard opening on item focus.
+ */
+export const CodeInput = ({
+  onChange,
+  accessibilityLabel,
+  testID,
+  onFocusedA11yView,
+  ...props
+}: CodeInputProps) => {
+  const hiddenInputRef = React.useRef<TextInput>(null);
+
+  useFocusEffect(
+    React.useCallback(() => {
+      const timeoutId = setTimeout(() => {
+        hiddenInputRef.current?.focus();
+      }, FOCUS_TIMEOUT_MS);
+
+      if (onFocusedA11yView) {
+        setAccessibilityFocus(onFocusedA11yView, A11Y_FOCUS_TIMEOUT_MS);
+      }
+
+      return () => {
+        clearTimeout(timeoutId);
+      };
+    }, [onFocusedA11yView])
+  );
+
+  return (
+    <View
+      accessible
+      accessibilityLabel={accessibilityLabel}
+      accessibilityElementsHidden
+    >
+      <TextInput
+        testID={testID}
```

Comment:
I'm afraid that if we don't have the possibility to open the keyboard also with a press on the `CodeInput` we could have edge-cases where, for some reason the virtual keyboard is not presented, the user is not able to enter the pin. What do you think?

Reply:
It seems to be an unhappy scenario! I think it's better to discuss about this problem and all the possible solutions to avoid it.